### PR TITLE
Stop using the boost version of tie and tuple by default

### PR DESCRIPTION
**Pull request by rgacogne**
### Short description

Also try to pull a bit less of boost headers everywhere in our code base. Hopefully it will make compilation a bit faster but I mostly hope to avoid using the boost version instead of the `std` one by default without knowing it.

This PR needs a good review, as the changes should not break anything but it might some subtle differences between the boost and `std` versions that I missed.

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: pdns/namespaces.hh [hunk 1]
```diff
@@ -21,7 +21,6 @@
  */
 #pragma once
 
-#include <boost/tuple/tuple.hpp>
 #include <boost/format.hpp>
```
Comment (omoerbeek) [inline pdns/namespaces.hh:24]: It would be nice to get rid of the remaining boost includes as well here and move them to the specific places where they are needed. But that 's likely better done in a separate PR.

**Review by omoerbeek — CHANGES_REQUESTED** [review]
Reads and builds fine on OpenBSD as well, but is not complete I think. 
`grep -lr 'tie(' *.[ch][c] | grep -v std::` gives me a few remaining hits in auth, rec and dnsdist, same for `tuple(`.

I'm as well a bit worried about potential subtle changes, but this is the right moment in the release cycles.

**Comment by rgacogne** [comment]
> Reads and builds fine on OpenBSD as well, but is not complete I think.

It depends on what our goal is :) My main goal was to remove the `using` lines making the boost version the one used by default:
```
using boost::make_tuple;
using boost::tuple;
using boost::tie;
```
and then I fixed the places that were no longer compiling.
I would be very happy to do more than that, though :)

**Comment by omoerbeek** [comment]
> > Reads and builds fine on OpenBSD as well, but is not complete I think.
> 
> It depends on what our goal is :) My main goal was to remove the `using` lines making the boost version the one used by default:
> 
> ```
> using boost::make_tuple;
> using boost::tuple;
> using boost::tie;
> ```
> 
> and then I fixed the places that were no longer compiling. I would be very happy to do more than that, though :)

My main issue is that it is unclear to which version "naked" `tie` and `tuple` refer to. Better be explicit.

**Review by omoerbeek — APPROVED** [review]
Added a small commit with three more cases
